lib/spack/spack/test/concretize.py

```diff
@@ -2913,3 +2913,38 @@ def test_spec_filters(specs, include, exclude, expected):
         factory=lambda: specs, is_usable=lambda x: True, include=include, exclude=exclude
     )
     assert f.selected_specs() == expected
+
+
+@pytest.mark.only_clingo("clingo only re-use feature being tested")
+@pytest.mark.regression("38484")
+def test_git_ref_version_can_be_reused(
+    install_mockery_mutable_config, do_not_check_runtimes_on_reuse
+):
+    first_spec = spack.spec.Spec("zlib-ng@git.2.1.5=2.1.5").concretized()
+    first_spec.package.do_install(fake=True, explicit=True)
+
+    with spack.config.override("concretizer:reuse", True):
+        second_spec = spack.spec.Spec("zlib-ng@git.2.1.5=2.1.5~opt").concretized()
+        # is_installed(first_spec)
+        assert second_spec.dag_hash() != first_spec.dag_hash()
+
+
+@pytest.mark.only_clingo("clingo only re-use feature being tested")
+def test_reuse_prefers_standard_over_git_versions(
```

Comment:
Have you confirmed that this consistently fails on the current develop? I would've thought we want to use a lower version for `standard_spec` to ensure that it would fail without the PR, but if it already consistently picks the wrong one then this is fine.

Reply:
I confirmed it failed when I wrote it. I have not tested it since. I didn't discover this failure mode until I fixed the issue with reuse.  Before this PR it would be impossible to expose this behavior due to the bug